README.md

```diff
@@ -1836,6 +1837,153 @@ default is <code>True</code>
 </table>
 
 
+<a id="#nixpkgs_python_repository"></a>
+
+### nixpkgs_python_repository
+
+<pre>
+nixpkgs_python_repository(<a href="#nixpkgs_python_repository-name">name</a>, <a href="#nixpkgs_python_repository-repository">repository</a>, <a href="#nixpkgs_python_repository-repositories">repositories</a>, <a href="#nixpkgs_python_repository-nix_file">nix_file</a>, <a href="#nixpkgs_python_repository-nix_file_deps">nix_file_deps</a>, <a href="#nixpkgs_python_repository-quiet">quiet</a>)
+</pre>
+
+Define a collection of python packages based on a nix file.
+
+The only entry point is a [`nix_file`](#nixpkgs_python_repository-nix_file)
+which should expose a `pkgs` and a `python` attributes. `python` is the
+python interpreter, and `pkgs` a set of python packages that will be made
+available to bazel.
+
+:warning: All the packages in `pkgs` are built by this rule. It is
+therefore not a good idea to expose something as big as `pkgs.python3` as
+provided by nixpkgs.
+
+This rule is instead intended to expose an ad-hoc set of packages for your
+project, as can be built by poetry2nix, mach-nix, dream2nix or by manually
+picking the python packages you need from nixpkgs.
+
+The format is generic to support the many ways to generate such packages
+sets with nixpkgs. See our python [`tests`](/testing/toolchains/python) and
+[examples](`/examples/toolchains/python`) to get started.
+
+This rule is intended to mimic as closely as possible the [rules_python
+API](https://github.com/bazelbuild/rules_python#using-the-package-installation-rules).
+`nixpkgs_python_repository` should be a drop-in replacement of `pip_parse`.
+As such, it also provides a `requirement` function.
+
+:warning: Using the `requirement` fucntion inherits the same advantages and
+limitations as the one in rules_python. All the function does is create a
+label of the form `@{nixpkgs_python_repository_name}//:{package_name}`.
+While depending on such a label directly will work, the layout may change
+in the future. To be on the safe side, define and import your own
+`requirement` function if you need to play with these labels.
+
+:warning: Just as with rules_python, nothing is done to enforce consistency
+between the version of python used to generate this repository and the one
+configured in your toolchain, even if you use nixpkgs_python_toolchain. You
+should ensure they both use the same python from the same nixpkgs version.
+While nixpkgs_python_repository could also register the toolchain used to
+generate the package set, this is not implemented yet.
```

Comment:
And inconsistencies would still be possible if another toolchain is registered next to it. So, it's unclear if we really want to do this.

Reply:
```suggestion
```
Maybe better to remove this here to not set up false expectations. Especially in lite of the upcoming bzlmod extensions, there may be better ways to deal with this in the bzlmod world.